virtcontainers/kata_agent.go

```diff
@@ -83,11 +83,25 @@ var (
 	maxHostnameLen       = 64
 )
 
+const (
+	agentTraceModeDynamic  = "dynamic"
+	agentTraceModeStatic   = "static"
+	agentTraceTypeIsolated = "isolated"
+	agentTraceTypeCollated = "collated"
+
+	defaultAgentTraceMode = agentTraceModeDynamic
+	defaultAgentTraceType = agentTraceTypeIsolated
+)
+
 // KataAgentConfig is a structure storing information needed
 // to reach the Kata Containers agent.
 type KataAgentConfig struct {
 	LongLiveConn bool
 	UseVSock     bool
+	Debug        bool
+	Trace        bool
```

Comment:
total nit, just curious: did you consider making these a sub-struct?

Reply:
For `Debug` and `Trace`? I'm not sure what that would buy us? Also this PR tries to fixup some of the inconsistencies in VC and we do already store debug+trace values for other objects like this.

Reply:
I meant for all enable, trace mode and trace type. 

This is fine.